state/runtime/evm/instructions_test.go

```diff
@@ -161,6 +396,939 @@ func TestMStore(t *testing.T) {
+		mockHost.On("SetStorage", mock.Anything, mock.Anything,
+			mock.Anything, mock.Anything).Return(runtime.StorageAdded).Once()
+
+		s.host = &mockHost
+
+		s.push(one)
+		s.push(zero)
+
+		opSStore(s)
+		assert.Equal(t, uint64(5000), s.gas)
+	})
+	t.Run("StorageDeleted", func(t *testing.T) {
+		s, closeFn := getState()
+		defer closeFn()
+
+		s.gas = 10000
+		s.config = &chain.ForksInTime{
+			Istanbul:       true,
+			Constantinople: true,
+		}
+		s.msg = &runtime.Contract{Address: types.StringToAddress("0x1")}
+
+		mockHost := mockHost{}
+		mockHost.On("SetStorage", mock.Anything, mock.Anything,
+			mock.Anything, mock.Anything).Return(runtime.StorageDeleted).Once()
+
+		s.host = &mockHost
+
+		s.push(one)
+		s.push(zero)
+
+		opSStore(s)
+		assert.Equal(t, uint64(5000), s.gas)
+	})
+}
+
+func TestBalance(t *testing.T) {
+	s, cancelFn := getState()
+	defer cancelFn()
+
+	t.Run("Istanbul", func(t *testing.T) {
+		gasLeft := uint64(300)
+		s.config = &chain.ForksInTime{
+			Istanbul: true,
+		}
+		mockHost := mockHost{}
+		mockHost.On("GetBalance", mock.Anything).Return(big.NewInt(100))
+
+		s.host = &mockHost
+		s.gas = 1000
+		opBalance(s)
+
+		assert.Equal(t, int64(100), s.pop().Int64())
+		assert.Equal(t, gasLeft, s.gas)
+	})
+
+	t.Run("EIP150", func(t *testing.T) {
+		gasLeft := uint64(600)
+		s.config = &chain.ForksInTime{
+			EIP150: true,
+		}
+		mockHost := mockHost{}
+		mockHost.On("GetBalance", mock.Anything).Return(big.NewInt(100))
+
+		s.host = &mockHost
+		s.gas = 1000
+		opBalance(s)
+
+		assert.Equal(t, int64(100), s.pop().Int64())
+		assert.Equal(t, gasLeft, s.gas)
+	})
+
+	t.Run("OtherForks", func(t *testing.T) {
+		gasLeft := uint64(980)
+
+		s.config = &chain.ForksInTime{
+			London: true,
+		}
+		mockHost := mockHost{}
+		mockHost.On("GetBalance", mock.Anything).Return(big.NewInt(100))
+
+		s.host = &mockHost
+		s.gas = 1000
+		opBalance(s)
+
+		assert.Equal(t, int64(100), s.pop().Int64())
+		assert.Equal(t, gasLeft, s.gas)
+	})
+}
+
+func TestSelfBalance(t *testing.T) {
+	t.Run("IstanbulFork", func(t *testing.T) {
+		s, cancelFn := getState()
+		defer cancelFn()
+
+		s.config = &chain.ForksInTime{
+			Istanbul: true,
+		}
+		mockHost := mockHost{}
+		mockHost.On("GetBalance", mock.Anything).Return(big.NewInt(100)).Once()
+
+		s.msg = &runtime.Contract{Address: types.StringToAddress("0x1")}
+		s.host = &mockHost
+		s.gas = 1000
+		opSelfBalance(s)
+
+		assert.Equal(t, int64(100), s.pop().Int64())
+	})
+
+	t.Run("NoForkErrorExpected", func(t *testing.T) {
+		s, cancelFn := getState()
+		defer cancelFn()
+
+		s.config = &chain.ForksInTime{}
+
+		mockHost := mockHost{}
+		mockHost.On("GetBalance", mock.Anything).Return(big.NewInt(100)).Once()
+
+		s.msg = &runtime.Contract{Address: types.StringToAddress("0x1")}
+		s.host = &mockHost
+		s.gas = 1000
+		opSelfBalance(s)
+
+		assert.True(t, s.stop)
+		assert.Equal(t, s.err, errOpCodeNotFound)
+	})
+}
+
+func TestChainID(t *testing.T) {
+	t.Run("IstanbulFork", func(t *testing.T) {
```

Comment:
Can you abstract common code accross the tests into some (anonymous) function(s) in order to reduce code repetitions accross different test cases?

For example you are always acquiring state instance, setting some initial gas, creating `mockHost` etc.